src/event.js

```diff
@@ -428,75 +398,40 @@ jQuery.event = {
 		return handlerQueue;
 	},
 
-	// Includes all common event props including KeyEvent and MouseEvent specific props
-	props: ( "altKey bubbles cancelable ctrlKey detail eventPhase " +
-		"metaKey shiftKey view which char charCode key keyCode " +
-		"button buttons clientX clientY offsetX offsetY pageX pageY screenX screenY toElement"
-	).split( " " ),
-
-	propHooks: {
-		which: function( event ) {
-			var button = event.button;
-
-			// Add which for key events
-			if ( event.which == null && rkeyEvent.test( event.type ) ) {
-				return event.charCode != null ? event.charCode : event.keyCode;
-			}
-
-			// Add which for click: 1 === left; 2 === middle; 3 === right
-			if ( !event.which && button !== undefined && rmouseEvent.test( event.type ) ) {
-				return ( button & 1 ? 1 : ( button & 2 ? 3 : ( button & 4 ? 2 : 0 ) ) );
-			}
-
-			return event.which;
-		},
-
-		pageX: function( event ) {
-			var eventDoc, doc, body;
-
-			// Calculate pageX if missing and clientX available
-			if ( event.pageX == null && event.clientX != null ) {
-				eventDoc = event.target.ownerDocument || document;
-				doc = eventDoc.documentElement;
-				body = eventDoc.body;
-
-				return event.clientX +
-					( doc && doc.scrollLeft || body && body.scrollLeft || 0 ) -
-					( doc && doc.clientLeft || body && body.clientLeft || 0 );
-			}
-
-			return event.pageX;
-		},
+	addProp: function( name, hook ) {
```

Comment:
> What kind of clobbering are you concerned about?

Ones where we decide to add a hook for some property in later versions but there might have already been previous attempts by others to fix the same problem and they overwrite our better solution after jQuery loads.

> Does this mean you're also convinced on jQuery.event.addProp, though?

I was going to say that if we're going to publicize `jQuery.Event.prototype` has the hooks we might as well just tell people to modify it themselves, but since the lazy hook code is tricky that's not a good idea. So yeah, I'll go for it. At minimum we need to fill `fixHooks` though via Migrate though. It's a rather obscure interface but people *were* using it.